headers/validation.h

```diff
@@ -108,6 +109,52 @@ auto operator|(T1 /*unused*/, T2 /*unused*/) {
 
 enum Separator { Space, Newline };
 
+//this contains some specific code which would be available in c++20
+namespace C20 {
+
+constexpr int countl_zero(unsigned long long x) {
+	int res = 64;
+	for (int i = 32; i > 0; i >>= 1) {
+		if ((x >> i) > 0) {
+			res -= i;
+			x >>= i;
+		}
+	}
+	if (x > 0) res--;
+	return res;
+}
+
+int popcount(unsigned long long x) {
+	return static_cast<int>(std::bitset<64>(x).count());
```

Comment:
Looks horrible, but I suppose `bitset<64>` is actually calling the platform dependant popcnt function so should be good :+1:

Reply:
that's what i hope for :D